tests/integration/conftest.py

```diff
@@ -121,6 +122,48 @@ def gateway_client_pod(client, worker_node, subnet_resource):
         client.delete(type(obj), obj.metadata.name, namespace=obj.metadata.namespace)
 
 
+@pytest.fixture()
+async def isolated_subnet(client: Client):
+    log.info("Creating isolated subnet resources ...")
+    path = Path("tests/data/isolated-subnet.yaml")
+    for obj in codecs.load_all_yaml(path.read_text()):
+        client.create(obj)
+    # Wait for the subnet and pods to become stable.
+    await asyncio.sleep(60)
```

Comment:
In my experience, fixed duration sleeps like this in tests usually lead to problems - we find that 60 seconds is *usually*, but not always, long enough. And we need the tests to be reliable, so we bump it up to 120 seconds, then 240 seconds, and so on. We end up slowing the tests down a lot more than they need for a typicial run, just to get 100% reliability that allows for outlier runs.

Instead of sleeping for 60 seconds, I recommend waiting for the specific condition that you need. I see below that the test waits for the pods to enter Ready condition - does that cover the need here already?